common/retry_handler_delays.go

```diff
@@ -0,0 +1,166 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	commontypes "github.com/agglayer/aggkit/common/types"
+	"github.com/agglayer/aggkit/config/types"
+)
+
+var (
+	// If this error is in the returned error, it means that the function should be aborted
+	// and no more retries should be attempted.
+	ErrAbort          = fmt.Errorf("abort")
+	ErrInvalidConfig  = fmt.Errorf("invalid config")
+	ErrExecutionFails = fmt.Errorf("execution fails after retries exceeded")
+)
+
+const (
+	MaxAttemptsInfinite = -1 // MaxAttemptsInfinite means infinite retries are allowed
+)
+
+// RetryHandlerDelays is a struct that holds the retry delays and the maximum number of retries.
+// It implements the RetryDelayer interface, which allows executing a function with retry logic.
+// The delays are specified as a slice of types.Duration, and the maximum number of retries can be set.
+// If maxRetries is set to 0, it means infinite retries are allowed.
+//
+// You can also abort the retrying wrapping the error ErrAbort into the result
+// this is useful if there are some conditions that should not be retried
+type RetryHandlerDelays struct {
+	RetryDelaysConfig
+}
+
+// NewRetryHandler creates a new RetryHandler with the specified delays and maximum retries.
+func NewRetryHandler(delays []types.Duration, maxRetries int) *RetryHandlerDelays {
+	return &RetryHandlerDelays{
+		RetryDelaysConfig: RetryDelaysConfig{
+			Delays:     delays,
+			MaxRetries: maxRetries,
+		},
+	}
+}
+
+// Validate checks if the RetryDelays configuration is valid.
+func (r *RetryHandlerDelays) Validate() error {
+	return r.RetryDelaysConfig.Validate()
+}
+
+// String returns a string representation of the RetryDelays struct.
+func (r *RetryHandlerDelays) String() string {
+	if r == nil {
+		return "RetryHandlerDelays{nil}"
+	}
+	return fmt.Sprintf("RetryHandlerDelays{%s}", r.RetryDelaysConfig.String())
+}
+
+// InfiniteRetries return true if the configuration allows infinite retries.
+func (r *RetryHandlerDelays) InfiniteRetries() bool {
+	// Infinite retries are allowed if MaxAttempts is 0.
+	return r != nil && r.MaxRetries == MaxAttemptsInfinite
+}
+
+func (r *RetryHandlerDelays) NoRetries() bool {
```

Comment:
This explains a bit more precise what this returns. `NoRetries` to me seems like you want it to have no retries, and not asking it if it is configured to have 0 retries.

```suggestion
func (r *RetryHandlerDelays) IsNoRetriesConfigured() bool {
```

Reply:
Also renamed  `InfiniteRetries()` to `IsInfiniteRetriesConfigured()`